src/components/PaymentRequestButtonElement.js

```diff
@@ -0,0 +1,115 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import shallowEqual from '../utils/shallowEqual';
+import type {ElementContext} from './Elements';
+
+type Props = {
+  className: string,
+  elementRef: Function,
+  onBlur: Function,
+  onFocus: Function,
+  onReady: Function,
+  paymentRequest: {
+    canMakePayment: Function,
+    on: Function,
+    show: Function,
+  },
+};
```

Comment:
I imagine we're using `Function` here because that's how it was done in the old `Element` component. We *could* get more type safety using Flow's `SyntheticEvent<T>` types: <https://flow.org/en/docs/react/events/>. Not sure if this is important enough to block.

Reply:
Hmm, but the events are not DOM Events, but events from the wrapped Stripe Element component. We could still be more specific, though! I'll leave that for a separate PR.